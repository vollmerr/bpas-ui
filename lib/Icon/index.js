var _templateObject = _taggedTemplateLiteral(['\n  font-size: ', 'px;\n'], ['\n  font-size: ', 'px;\n']);

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import './style.css';

var StyledIcon = styled.span(_templateObject, function (props) {
    return props.size;
});

/** Renders an icon */
function Icon(_ref) {
    var icon = _ref.icon,
        className = _ref.className,
        _ref$size = _ref.size,
        size = _ref$size === undefined ? 20 : _ref$size,
        props = _objectWithoutProperties(_ref, ['icon', 'className', 'size']);

    return React.createElement(StyledIcon, Object.assign({
        className: 'ca-gov-icon-' + icon + ' ' + className,
        size: size
    }, props));
}

Icon.propTypes = {
    icon: PropTypes.string.isRequired,
    className: PropTypes.string,
    size: PropTypes.number
};

export default Icon;