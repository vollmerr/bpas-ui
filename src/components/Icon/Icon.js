import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

/** Renders an icon */
function Icon({
    icon,
    className,
    ...props,
}) {
    return <span className={`ca-gov-icon-${icon} ${className}`} {...props}/>;
}

Icon.propTypes = {
    icon: PropTypes.string.isRequired,
    className: PropTypes.string,
};

export default Icon;
