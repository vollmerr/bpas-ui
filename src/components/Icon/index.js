import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import './style.css';

const StyledIcon = styled.span`
  font-size: ${(props) => props.size}px;
`;

/** Renders an icon */
function Icon({
    icon,
    className,
    size = 20,
    ...props,
}) {
    return (<StyledIcon
      className={`ca-gov-icon-${icon} ${className}`}
      size={size}
      {...props}
    />);
}

Icon.propTypes = {
    icon: PropTypes.string.isRequired,
    className: PropTypes.string,
    size: PropTypes.number,
};

export default Icon;
