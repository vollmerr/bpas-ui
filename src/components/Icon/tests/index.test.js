import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../';

const testIcon = 'home';

describe('<Icon />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Icon icon={testIcon} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('passes the correct props down', () => {});
});
