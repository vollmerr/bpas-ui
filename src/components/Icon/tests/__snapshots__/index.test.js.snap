// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly 1`] = `
<span
  className="ca-gov-icon-home undefined sc-bdVaJa iHmnvJ"
  size={20}
/>
`;
